script.js

```diff
@@ -0,0 +1,164 @@
+            {text: "Poland", isCorrect: false},
+            {text: "Italy", isCorrect: false},
+            {text: "France", isCorrect: false},
+        ]
+    },
+    {
+        text: "Which is largest continent in the world?",
+        answers: [
+            {text: "Asia", isCorrect: true},
+            {text: "Africa", isCorrect: false},
+            {text: "Australia", isCorrect: false},
+            {text: "Europe", isCorrect: false},
+            ]
+    },
+    {
+        text: "Do I love my handsome boy",
+        answers: [
+            {text: "Yes", isCorrect: true},
+            {text: "No", isCorrect: false},
+        ]
+    },
+    {
+        text: "Which is the longest river in the world?",
+        answers: [
+            {text: "Congo", isCorrect: false},
+            {text: "Yellow", isCorrect: false},
+            {text: "Nile", isCorrect: true},
+            {text: "Niger", isCorrect: false},
+            ]
+    }
+];
+const QUESTION_ELEMENT = document.getElementById("question");
+const ANSWER_BUTTONS_CONTAINER_ELEMENT = document.getElementById("answer-buttons");
+const NEXT_BUTTON_ELEMENT = document.getElementById("next-btn");
+const CORRECT_ANSWER_CLASS = "correct";
+const INCORRECT_ANSWER_CLASS = "incorrect";
+const ANSWER_HTML_TAG_NAME = "button";
+const ANSWER_CLASSES = "btn";
+
+
+let currentQuestionIndex = 0;
+let score = 0;
+
+function startQuiz(){
+    currentQuestionIndex = 0;
+    score = 0;
+    NEXT_BUTTON_ELEMENT.innerHTML = "Next";
+    showQuestion();
+}
+
+function showQuestion(){
+    resetState();
+    let currentQuestion = QUESTIONS[currentQuestionIndex];
+    let questionNo = currentQuestionIndex + 1;
+    QUESTION_ELEMENT.innerHTML = questionNo + ". " + currentQuestion.text;
+
+    currentQuestion.answers.forEach(answer => prepareAnswer(answer));
+}
+
+function prepareAnswer(answer) {
+    const answerElement = createAnswerElement(answer);
+    ANSWER_BUTTONS_CONTAINER_ELEMENT.appendChild(answerElement);
+}
+
+function createAnswerElement(answer) {
+    const answerElement = document.createElement(ANSWER_HTML_TAG_NAME);
+    answerElement.innerHTML = answer.text;
+    answerElement.classList.add(ANSWER_CLASSES);
+    answerElement.dataset.isCorrect = answer.isCorrect;
+    answerElement.addEventListener("click", handleOnClickAnswer);
+
+    return answerElement;
+}
+
+function resetState() {
+    NEXT_BUTTON_ELEMENT.style.display = "none";
+    removeAllChildrenOf(ANSWER_BUTTONS_CONTAINER_ELEMENT);
+}
+
+function removeAllChildrenOf(element) {
+    while(element.firstChild) {
+        element.removeChild(element.firstChild);
+    }
+}
+
+function handleOnClickAnswer(e){
+    const selectedButtonElement = e.target;
+    const isCorrect = selectedButtonElement.dataset.isCorrect === "true";
+
+    const extraAnswerClass = isCorrect ? CORRECT_ANSWER_CLASS : INCORRECT_ANSWER_CLASS;
+    selectedButtonElement.classList.add(extraAnswerClass);
+
+    if(isCorrect) {
+        score++;
+    }
```

Comment:
ok there is too much of code in the function...The 102-109 lines may be in extra function called `processSelectedButtonElement`:

```javascript
function handleOnClickAnswer(e){
    const selectedButtonElement = e.target;
    const answerButtons = Array.from(ANSWER_BUTTONS_CONTAINER_ELEMENT.children); // PLEASE NOTICE ME!!! I WAS MOVED! :D
    
    processSelectedButtonElement(selectedButtonElement);
    prepareButtonsAfterAnswer(answerButtons);
}

function processSelectedButtonElement(buttonElement) {
    const isCorrect = buttonElement.dataset.isCorrect === "true";

    const extraAnswerClass = isCorrect ? CORRECT_ANSWER_CLASS : INCORRECT_ANSWER_CLASS;
    buttonElement.classList.add(extraAnswerClass);

    if(isCorrect) {
        score++;
    }
}
```

Reply:
Hm ok but now `processSelectedButtonElement` has too much... so we may create this:

```javascript
function processSelectedButtonElement(buttonElement) {
    const isCorrect = buttonElement.dataset.isCorrect === "true";

    applyAnswerClass(buttonElement, isCorrect);
    updateScore(isCorrect);
}

function applyAnswerClass(buttonElement, isCorrect) {
    const extraAnswerClass = isCorrect ? CORRECT_ANSWER_CLASS : INCORRECT_ANSWER_CLASS;
    buttonElement.classList.add(extraAnswerClass);
}

function updateScore(isCorrect) {
    if(isCorrect) {
        score++;
    }
}
```

Reply:
Ok now is much better but what if do not want to pass isCorrect to applyAnswerClass function? The information whether the selected button is correct contains in buttonElement already! Well we may create function that based on buttonElement object checks if it is correct :). Let's see it:

```javascript
function processSelectedButtonElement(buttonElement) {
    applyAnswerClass(buttonElement);
    updateScore(buttonElement);
}

function applyAnswerClass(buttonElement) {
    const extraAnswerClass = isButtonElementCorrect(buttonElement) ? CORRECT_ANSWER_CLASS : INCORRECT_ANSWER_CLASS;
    buttonElement.classList.add(extraAnswerClass);
}

function updateScore(buttonElement) {
    if(isButtonElementCorrect(buttonElement)) {
        score++;
    }
}


// "magic" function ;P it is really simple, but it makes code pure and lovely <3
function isButtonElementCorrect(buttonElement) {
    return buttonElement.dataset.isCorrect === "true"
}
```

Reply:
Done.